src/updater/updater.cpp

```diff
@@ -408,27 +474,98 @@ updateTurnPenalties(const UpdaterConfig &config,
 
         if (turn_weight_penalty < 0)
         {
-            util::Log(logWARNING) << "Negative turn penalty at " << turn_index.from_id << ", "
-                                  << turn_index.via_id << ", " << turn_index.to_id
+            util::Log(logWARNING) << "Negative turn penalty at " << internal_turn.from_id << ", "
+                                  << internal_turn.via_id << ", " << internal_turn.to_id
                                   << ": turn penalty " << turn_weight_penalty;
         }
     }
 
     return updated_turns;
 }
+std::vector<std::uint64_t>
+updateConditionalTurns(const UpdaterConfig &config,
+                       std::vector<TurnPenalty> &turn_weight_penalties,
+                       const std::vector<extractor::TurnRestriction> &conditional_turns,
+                       const std::vector<extractor::QueryNode> &internal_to_external_node_map,
+                       Timezoner time_zone_handler)
+{
+    const auto turn_index_region =
+        mmapFile(config.turn_penalties_index_path, boost::interprocess::read_only);
+    // Mapped file pointer for turn indices
+    const extractor::lookup::TurnIndexBlock *turn_index_blocks =
+        reinterpret_cast<const extractor::lookup::TurnIndexBlock *>(
+            turn_index_region.get_address());
+    BOOST_ASSERT(is_aligned<extractor::lookup::TurnIndexBlock>(turn_index_blocks));
+
+    std::vector<std::uint64_t> updated_turns;
+    if (conditional_turns.size() == 0)
+        return updated_turns;
+
+    // TODO make this into a function
+    LookupTable<std::tuple<NodeID, NodeID>, NodeID> is_only_lookup;
+    std::unordered_set<std::tuple<NodeID, NodeID, NodeID>,
+                       std::hash<std::tuple<NodeID, NodeID, NodeID>>>
```

Comment:
This doesn't work, boost::hash implementation for tuples doesn't work across all the systems we test on https://github.com/Project-OSRM/osrm-backend/issues/3962#issuecomment-297096071